Signal/src/ViewControllers/ConversationView/Cells/OWSMessageCell.m

```diff
@@ -263,8 +265,120 @@ - (void)loadForDisplay:(int)contentWidth
     //    });
 }
 
-- (void)updateDateHeader:(int)contentWidth
+- (nullable id)tryToLoadCellMedia:(nullable id (^)())loadCellMediaBlock mediaView:(UIView *)mediaView
```

Comment:
We now eagerly create out view hierarchy (to do this exactly once per cell usage) but lazy-load any expensive media (photo, gif, etc.) used in those views.  Note that this lazy-load can fail, in which case we modify the view hierarchy to use an "error" state.  The `didCellMediaFailToLoad` reflects media load fails.